src/api/rpc_api/runtime_update.rs

```diff
@@ -0,0 +1,72 @@
+/*
+	Copyright 2023 Supercomputing Systems AG
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+		http://www.apache.org/licenses/LICENSE-2.0
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+*/
+use crate::{api::Error, rpc::Subscribe, rpc_api::EventSubscriptionFor, Result};
+use alloc::sync::Arc;
+use codec::Decode;
+use core::sync::atomic::{AtomicBool, Ordering};
+use serde::de::DeserializeOwned;
+
+/// Struct to support waiting for runtime updates
+pub struct RuntimeUpdateDetector<Hash, Client>
+where
+	Hash: DeserializeOwned + Copy + Decode,
+	Client: Subscribe,
+{
+	subscription: EventSubscriptionFor<Client, Hash>,
```

Comment:
Regarding the problem with the cancellation token: Subscription allows for "unsubscribe":

https://github.com/scs/substrate-api-client/blob/4d4f19810f607371cf84e5885f398ab483f5aec9/src/rpc/mod.rs#L68-L81

Maybe I'm imagining things a little easier than they are, but: If the subscription is made public, or a respective function is implemented, one could unsubscribe, and break the loop by having `.next_events_from_metadata()` return `None`. That would resolve the problem with the sync version and make the cancellation token obsolete. Could that work?

Reply:
Hmm, it might work. I'm not quite sure what happens with waiting subscriptions when we unsubscribe. Anyway I think I would rather look at this in a follow up task.

Reply:
Agreed, lets investigate that in another issue

Reply:
I created #638 as a follow up